src/PlaywightSharp.ProtocolTypesGenerator/ChromiumProtocolTypesGenerator.cs

```diff
@@ -0,0 +1,400 @@
+                            _knownTypes[$"{domain.Domain}.{type.Id}"] = itemType + "[]";
+                        }
+                    }
+                    else if (type.Type == "string" && type.Enum == null)
+                    {
+                        _knownTypes[type.Id] = "string";
+                        _knownTypes[$"{domain.Domain}.{type.Id}"] = "string";
+                    }
+                    else if (type.Type == "integer")
+                    {
+                        _knownTypes[type.Id] = "int";
+                        _knownTypes[$"{domain.Domain}.{type.Id}"] = "int";
+                    }
+                    else if (type.Type == "number")
+                    {
+                        _knownTypes[type.Id] = "double";
+                        _knownTypes[$"{domain.Domain}.{type.Id}"] = "double";
+                    }
+                }
+            }
+
+            // work around, too lazy to solve this
+            _knownTypes["Headers"] = "System.Collections.Generic.IDictionary<string, string>";
+            _knownTypes["ArrayOfStrings"] = "int[]";
+            _knownTypes["StringIndex"] = "int";
+        }
+
+        private void GenerateTypes(StringBuilder builder, ChromiumProtocolDomain domain)
+        {
+            if (domain.Types == null)
+            {
+                return;
+            }
+
+            foreach (var type in domain.Types)
+            {
+                if (type.Enum != null)
+                {
+                    builder.AppendLine("/// <summary>");
+                    builder.Append("/// ").AppendLine(FormatDocs(type.Description));
+                    builder.AppendLine("/// </summary>");
+                    builder.Append("internal enum ").AppendLine(type.Id);
+                    builder.AppendLine("{");
+                    builder.AppendJoin(",\n", NormalizeEnum(type.Enum));
+                    builder.AppendLine("}");
+                }
+                else if (type.Type == "object")
+                {
+                    builder.AppendLine("/// <summary>");
+                    builder.Append("/// ").AppendLine(FormatDocs(type.Description));
+                    builder.AppendLine("/// </summary>");
+                    builder.Append("internal class ").AppendLine(type.Id);
+                    builder.AppendLine("{");
+                    builder.AppendJoin("\n", NormalizeProperties(type.Properties));
+                    builder.AppendLine("}");
+                }
+            }
+        }
+
+        private void GenerateCommands(StringBuilder builder, ChromiumProtocolDomain domain)
+        {
+            if (domain.Commands == null)
+            {
+                return;
+            }
+
+            foreach (var command in domain.Commands)
+            {
+                // request
+                string baseName = $"{domain.Domain}{char.ToUpper(command.Name[0])}{command.Name.Substring(1)}";
+                builder.AppendLine("/// <summary>");

```

Comment:
Let's treat warnings as errors.

Reply:
is warning that I'm not using the result. just calling `AppendLine` modifies the StringBuilder. there's no reason to use the result here...
in this case, the Analyzers are not helping at all

Reply:
Yeah, that rule is quite silly. You can disable that one across the board.